### PR TITLE
Screen backlight timer

### PR DESCRIPTION
**Pull request by nolith**
This is a WIP, right now it send the screen in standby after 30s idle.

TODO:
- [x] get out from standby when the squelch open
- [x] get out from standby when in TX (PPT is not a keyboard event)
- [x] add a menu to configure the desired timeout (also no timeout)
- [x] store the timeout value in the settings

### CODE REVIEW

**Comment by silseva** [comment]
In the next days I'll add an `rtx_squelchStatus()` function returning true when the squelch is open.
Meanwhile I'm thinking where is the best place to put the backlight logic: probably we can put it temporarily in the keyboard task because of its 40Hz update rate.
As a general implementation I'd put a downcounter in the global state, decremented by one at each round by the state task, which ticks at 1Hz. PTT press and squelch opening turn on the backlight and set the counter to the timeout value then, when it reaches zero, the backlight is turned off.

**Comment by nolith** [comment]
Something that I realized using this code on a real QSO is that, like in all the other radio, you don't want the key press that get you out of the standby to be actionable. It's only purpose is to light on the screen

**Comment by silseva** [comment]
Another point in favor of putting the backlight logic inside the keyboard task: in this way you can "trap" the keypress event and do not transmit it to the UI thread if the backlight is off. That is, the first press on any button with backlight off only turns on the backlight.

**Comment by nolith** [comment]
@silseva I'll keep working on my other items from the list, but I think that even in the current shape is a good improvement.

**Comment by silseva** [comment]
I added the `rtx_rxSquelchOpen()` API function, allowing to query the current status of the RX audio squelch

File: openrtx/src/ui/ui.c [hunk 1]
```diff
@@ -381,6 +385,7 @@ layout_t _ui_calculateLayout()
 
 void ui_init()
 {
+    last_event_tick = getTick();
```
Comment (nolith) [inline openrtx/src/ui/ui.c:388]: This is needed otherwise the linux emulator will start in standby

File: openrtx/src/ui/ui.c [hunk 2]
```diff
@@ -640,6 +645,27 @@ void _ui_changeContrast(int variation)
     display_setContrast(state.settings.contrast);
 }
 
+void _ui_enterStandby()
+{
+    if(standby)
+        return;
```
Comment (nolith) [inline openrtx/src/ui/ui.c:701]: Is this allowed under Allman bracing or should I always use {} even with just a return?

**Comment by nolith** [comment]
> I added the `rtx_rxSquelchOpen()` API function, allowing to query the current status of the RX audio squelch

Thank you 🙏 

I'll try to find some time to work on the settings, both UI and memory.

File: openrtx/src/ui/ui.c (hunk 2 above)
Comment (silseva) [reply openrtx/src/ui/ui.c:701]: Keep it as it is now, we're not super-strict about those things!

**Comment by silseva** [comment]
> > I added the `rtx_rxSquelchOpen()` API function, allowing to query the current status of the RX audio squelch
> 
> Thank you pray
> 
> I'll try to find some time to work on the settings, both UI and memory.

Sorry for the long processing time! Even if it was a 5-minute thing, it slipped out from my mind for a while...

File: openrtx/include/settings.h [hunk 3]
```diff
@@ -23,15 +23,37 @@
...
-    bool gps_enabled;       // GPS active
-    char callsign[10];      // Plaintext callsign, for future use
+    uint8_t brightness;                  // Display brightness
+    uint8_t contrast;                    // Display contrast
+    uint8_t sqlLevel;                    // Squelch level
+    uint8_t voxLevel;                    // Vox level
+    int8_t utc_timezone;                 // Timezone
+    bool gps_enabled;                    // GPS active
+    char callsign[10];                   // Plaintext callsign, for future use
+    display_timer_t brightness_timer;    // Standby timer
```
Comment (nolith) [inline openrtx/include/settings.h]: will `__attribute__((packed))` be enough for this to 4bits only?

i guess no, I think this needs the union trick (like in event.h)

File: openrtx/include/settings.h [hunk 4]
```diff
@@ -23,15 +23,37 @@
...
+    TIMER_30S =  6,
+    TIMER_1M  =  7,
+    TIMER_2M  =  8,
+    TIMER_3M  =  9,
+    TIMER_4M  = 10,
+    TIMER_5M  = 11,
+    TIMER_15M = 12,
+    TIMER_30M = 13,
+    TIMER_45M = 14,
+    TIMER_1H  = 15
```
Comment (nolith) [inline openrtx/include/settings.h:43]: these are the values I have on my anytone, and they nicely fits into 4 bits.

**Review by nolith — COMMENTED** [review]
I'll do a final review once I have the radio.

It works on the emulator.

I'll probably do a round of refactoring to harmonise names, but the overall feature is ready.

**Comment by nolith** [comment]
@silseva I think this is ready for review now

**Pull request merged**